### ISSUE TITLE
"Other" fields

### ISSUE THREAD

**Issue by bhaugen** [FreedomCoop/valuenetwork#64]
Requested so far for skills on worker profiles, to alert admins that somebody has suggested something to be created.

Add other places for "Other" fields here.

**Comment by skoria** [comment]
In the worker app, perhaps when logging non -process work they can suggest other types of work. But I'm not sure if this can also relate to the skills. Mostly for non process work though, each project must be able to define it's own types of work. Maybe by adding the ability to add another map location for users, they can also change their address using the already existing address geocoding functionality

**Comment by skoria** [comment]
Post launch

**Comment by XaviP** [comment]
Solved. Closing. Please, reopen if needed.

**Comment by bum2** [comment]
We need this feature asap also, at least for the Skills in the profile page of workapp. The user proposed 'skills' should be modeated and created by admins, so we need a workflow for the 'Other' skills feature. Initially most of the 'new skill requests' will be simply added, but sometimes the admins can see no need to add a new skill (if a good similar skill is already available), and this is the reason of moderating the user proposals and don't make them automatic (also to assure a common writing style).
I think the best approach for Skills (perhaps for the future) is to make them nestable (hierachical), so they can be grouped (and navigated) by knowledge areas, sectors and specialities.

**Comment by bhaugen** [comment]
> I think the best approach for Skills (perhaps for the future) is to make them nestable (hierachical), so they can be grouped (and navigated) by knowledge areas, sectors and specialities.

Skills can be nested by faceted classifications (see on the inventory and resource types pages), or they can have parents (be hierarchical). Or both.

**Comment by fosterlynn** [comment]
Agree with need for moderation of skills listings.

>Skills can be nested by faceted classifications (see on the inventory and resource types pages), or they can have parents (be hierarchical). Or both.

Will need to figure out, and upgrade the UI though.

**Comment by bum2** [comment]
Marvelous! the point here is start to design a workflow to let users propose 'other' skills... i can think about the UI, but the internal system and moderation of new skills is waiting someone with more ocp experience than me... Which will be the best way to do it?
Admins can take care of facets or setting the parent of each skill, but a list of proposed new skills (with info of the user requesting them) will be good for admins to prioritize and configure. 
The communication can be done with the existent comments thread, so we can start there the process (users use the comments to ask for new skills) and meanwhile we think a better way.
At UI level, i was imagining just a simple TextField below the list of skills in the profile page, with the label 'Other' and the input text. When user clicks 'save', if there's content in this field it become added to a list somewhere, for admins or for 'managers' of a 'Skills Structure Team'. Then admins can create and nest correctly the skill proposals and then erase it from the list.

**Comment by bhaugen** [comment]
> Which will be the best way to do it?

I think a new model in the work app:
* SkillSuggestion (or something like that)
   * skill
   * description
   * suggested_by
   * suggestion_date

What else do you want to know?

**Comment by bum2** [comment]
is ok! description not needed.

**Comment by bhaugen** [comment]
First draft of a model for this:
```
class SkillSuggestion(models.Model):
    skill = models.CharField(_('skill'), max_length=128,
        help_text=_("A new skill that you want to offer that is not already listed")))
    suggested_by = models.ForeignKey(User, verbose_name=_('suggested by'),
        related_name='skill_suggestion', blank=True, null=True, editable=False)
    suggestion_date = models.DateField(auto_now_add=True, blank=True, null=True, editable=False)
```

Please suggest improvements.

**Comment by bhaugen** [comment]
I think SkillSuggestion needs more thought and work. It is the start of a conversation. A person's skill suggestions should be visible on their profile along with the skills they have already selected. Then if and when a new EconomicResourceType is created following their suggestion, they should be notified in some way. Maybe just a note that appears on their profile page next to the newly added skill saying something like "thanks for the suggestion".

But that requires maybe 3 new fields on SkillSuggestion: resource_type, created_by, and created_date. The new skill suggestions should appear to coop admins somewhere, too. 

But what do we do with rejected suggestions?

**Comment by bum2** [comment]
The conversation about this new skills can be done in the comments thread we have (it can be used and renamed as 'admins feedback' or similar, after register).
For admins i think we need a new page to manage that suggestions simply, but one day soon we will need also a page to build the tree of skills. There're some good jquery plugins to build trees with drag'n'drop, etc. Also in the profile, the skills part will need a tree navigating widget, also a jquery plugin we will add some day. (I will propose both plugins if i find them in a old project i was working on, with trees, but that will be for another future release).
No needed a thanks note in the profile: admins can say so in a comment. Also the rejection arguments or the assimilation of a suggestion into another similar skill name already created... every communication in the same comments thread we have (related membership).

**Comment by bhaugen** [comment]
I am picking up in the chatter that this "Other" field will be used for other Resource Types besides skills, so I am renaming the model to ResourceTypeSuggestion. I hope that is ok with everybody. Best to do it now before it gets deployed.

**Comment by bhaugen** [comment]
I wrote a comment that I was renaming SkillSuggestion but then thought better about it. I think other resource type suggestions might have different behaviors and at least different help texts, so I'm leaving SkillSuggestion as-is for now.

**Comment by bhaugen** [comment]
Just pushed the first drop of the "Other" field to the Project branch:
https://github.com/FreedomCoop/valuenetwork/commit/2a162e1e0636b11bb2ef3ec1d11e179fbb969c50

The only place you can see the Skill Suggestions now is in django admin.

Screen shot from profile page:
![other_field](https://cloud.githubusercontent.com/assets/117439/18092794/5453af0a-6e93-11e6-8656-ca3b84932747.png)

**Comment by bum2** [comment]
Looks great Bob! I've pulled all, but when i go to the profile, it throws me an error:
`global name 'other_form' is not defined`
in work/views.py line 256

**Comment by bhaugen** [comment]
@bum2 what branch are you working in?

**Comment by bhaugen** [comment]
This is odd. The definition of other_form got deleted in your master merge fix here: 
https://github.com/FreedomCoop/valuenetwork/commit/c1a019989402f171a8e0a05b14301473ef82cdb8

Bunch of stuff deleted.

**Comment by bhaugen** [comment]
Ok, restored deleted lines, and pushed to the project branch.

**Comment by bhaugen** [comment]
@bum2 I'll leave this branch to you now, and stop working on it myself. If you need help on it, let me know.

**Comment by bhaugen** [comment]
@bum2 from your telegram chat, it looks like you are not following up with this branch. I saw something somewhere in telegram about wanting to take it through to admin creation of new skills (resource types). I'll think about how to do that and write a proposal here by tomorrow morning and then if it looks good to everybody, start working on it.

**Comment by bhaugen** [comment]
Now we need to figure out how to handle these new skill suggestions. Basically, a coop admin needs to review them and (maybe) turn them into new work resource types.
But they are also the start of a conversation with whoever suggested them, so they might need some comments, which could go on the worker's discussion page. And then maybe if the new skill suggestion is enacted, the worker could see a little "thanks for the suggestion" tag on their skills list on their profile?
 Also, should coop admins be notified about these, like membership requests?
 In other words, is the workflow here maybe a lot like membership requests? Or is that too much ceremony?
